hbase-server/src/main/java/org/apache/hadoop/hbase/quotas/QuotaUtil.java

```diff
@@ -390,7 +390,7 @@ protected static UserQuotaState buildDefaultUserQuotaState(Configuration conf) {
     buildDefaultTimedQuota(conf, QUOTA_DEFAULT_USER_MACHINE_WRITE_SIZE)
       .ifPresent(throttleBuilder::setWriteSize);
 
-    UserQuotaState state = new UserQuotaState();
+    UserQuotaState state = new UserQuotaState(EnvironmentEdgeManager.currentTime());
```

Comment:
I think we need to add nowTs as an argument to this method. During initial creation of the quota (for a new user), you'd pass 0. During refresh, you'd pass the nowTs that exists within the fetchUserQuotas method.

Reply:
The problem with always using currentTime() here is that when we add a new user to the quota map, we trigger an immediate refresh. The idea is that the 0 value woudl cause the new user to be refreshed. But if you set it to now, it wouldn't be refreshed until the next refresh interval.

Reply:
Oh yeah